src/olympia/abuse/models.py

```diff
@@ -134,6 +134,7 @@ def report(cls, abuse_report):
         )
         cinder_job, _ = cls.objects.get_or_create(job_id=job_id)
         abuse_report.update(cinder_job=cinder_job)
+        cls.get_entity_helper(abuse_report).report_additional_context()
```

Comment:
should we get the entity helper once and then use it for `report` and `report_additional_context`, or does that mess up the generators?

Reply:
I've implemented that in f6992c0938acf8c197c4cd08a928cbcaabb541ef and stored the queryset the generator is using in a property. That way we still get a new generator every time, but the queryset is cached by Django when evaluated, so re-using the same property (computed at entity init time) avoids the duplicated query.